bee-message/src/output/mod.rs

```diff
@@ -159,10 +163,58 @@ impl Output {
         }
     }
 
-    /// Verify if a valid storage deposit was made. Each [`Output`] has to have an amount that covers its associated
-    /// byte cost, given by [`ByteCostConfig`]. If there is a
-    /// [`StorageDepositReturnUnlockCondition`](unlock_condition::StorageDepositReturnUnlockCondition), its amount
-    /// is also checked.
+    /// Returns the chain identifier of an [`Output`], if any.
+    pub fn chain_id(&self) -> Option<ChainId> {
+        match self {
+            #[cfg(feature = "cpt2")]
+            Self::SignatureLockedSingle(_) => None,
+            #[cfg(feature = "cpt2")]
+            Self::SignatureLockedDustAllowance(_) => None,
+            Self::Treasury(_) => None,
+            Self::Basic(_) => None,
+            Self::Alias(output) => Some(output.chain_id()),
+            Self::Foundry(output) => Some(output.chain_id()),
+            Self::Nft(output) => Some(output.chain_id()),
+        }
+    }
+
+    ///
+    pub fn state_transition(
```

Comment:
Wouldn't `validate_state_transition` be a better and more consistent name. It sets the expectation straight what to expect of this method, and erroneously convey that state gets mutated.

Reply:
I'll call it `verify_state_transition` to be consistent with our other verification functions